### PR TITLE
Adding ESLint

### PR DESCRIPTION
**Pull request by SamKeathley**
Per issue #27 and some discussion there, add linting to project to fix some current styling problems and for stylistic consistency in the future. Eslint-config-standard has some good starting point configurations for us that we can add onto or change. I have also taken care of the current linting problems after getting all of this set up.

Eslint-config-standard offers some shared configs for us to start out with to get our code looking pretty. I also implemented the rollup-eslint-plugin since we are using rollup already and it claims to verify entry point and all imported files with ESLint.

####Learning
[Eslint-config-standard](https://github.com/standard/eslint-config-standard#eslint-config-standard----)
[Rollup-esling-plugin](https://github.com/TrySound/rollup-plugin-eslint#rollup-plugin-eslint-)

### CODE REVIEW

File: packages/convergence/package.json [hunk 1]
```diff
@@ -19,6 +19,14 @@
     "chai-as-promised": "^7.1.1",
     "mocha": "^4.0.1",
     "rollup": "^0.53.0",
-    "rollup-plugin-babel": "^4.0.0-beta.0"
+    "rollup-plugin-babel": "^4.0.0-beta.0",
+    "rollup-plugin-eslint": "^4.0.0"
+  },
+  "dependencies": {
```
Comment (cowboyd) [inline packages/convergence/package.json]: Are these `dependencies` or `devDependencies`?

File: packages/convergence/src/converge-on.js [hunk 2]
```diff
@@ -23,7 +23,7 @@
  * @returns {Promise} resolves if the assertion passes at least once;
  * if `always` is true, then rejects at the first error instead
  */
-export default function convergeOn(assertion, timeout = 2000, always, useStats) {
+export default function convergeOn (assertion, timeout = 2000, always, useStats) {
```
Comment (cowboyd) [inline packages/convergence/src/converge-on.js]: I don't know how I feel about this rule. Seems like not the way most setups work?

File: packages/convergence/package.json (hunk 1 above)
Comment (SamKeathley) [reply packages/convergence/package.json]: ```devDependencies```

File: packages/convergence/src/converge-on.js (hunk 2 above)
Comment (SamKeathley) [reply packages/convergence/src/converge-on.js]: I will change that.

File: packages/convergence/package.json (hunk 1 above)
Comment (SamKeathley) [reply packages/convergence/package.json]: They are ```devDependencies```

**Comment by wwilsman** [comment]
Thanks for taking care of this!

I was thinking it'd be nice if we could do `yarn lint` and just get linting results without having to do the entire build. Also, we probably want to lint our tests as well, which don't get included with the rollup bundle. So maybe we don't need `rollup-eslint-plugin` after all?

Also, I noticed some qunit things slipped into this branch. 😉

**Pull request closed without merging**